netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -99,7 +100,7 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
         if (evt instanceof SslHandshakeCompletionEvent) {
           SslHandshakeCompletionEvent handshakeEvent = (SslHandshakeCompletionEvent) evt;
           if (handshakeEvent.isSuccess()) {
-            if (sslHandler(ctx).applicationProtocol() != null) {
+            if (GrpcSslContexts.HTTP2_VERSIONS.contains(sslHandler(ctx).applicationProtocol())) {
```

Comment:
Need the same fix for client-side.  See every use of `applicationProtocol` in this file.